.github/workflows/release.yml

```diff
@@ -5,12 +5,43 @@ on:
     tags: ["v*"]
 
 jobs:
-  github_release:
+  log-updates:
+    name: Log packages to publish
+    runs-on: ubuntu-latest
+    steps:
+      - name: Checkout the new tag
+        uses: actions/checkout@v2
+        with:
+          fetch-depth: 2
+      - name: This release will publish the following packages
+        run: git diff --name-only HEAD^..HEAD
+
+  npm-release:
+    name: Publish release on npm
+    runs-on: ubuntu-latest
+    needs: log-updates
+    environment: npm
+    steps:
+      - name: Checkout the new tag
+        uses: actions/checkout@v2
+        with:
+          fetch-depth: 0
+      - name: Build and Test
+        run: make prepublish
+      - name: Publish to npm
+        run: yarn release-tool publish --yes
```

Comment:
Note: This line is the only untested step in this PR (I commented it out, run the action, and everything worked). However, we test in in the e2e tests on CircleCI where we publish to verdaccio.